packages/router/src/util.ts

```diff
@@ -31,13 +31,23 @@ export interface ParamType {
 }
 
 /** Definitions of the core param types. */
-const coreParamTypes: Record<string, ParamType> = {
+const coreParamTypes = {
   Int: {
     constraint: /\d+/,
     transform: Number,
   },
+  Float: {
+    constraint: /[+-]?[0-9]*[.]?[0-9]+/,
+    transform: Number,
+  },
+  Boolean: {
+    constraint: /true$|false$/,
```

Comment:
```suggestion
    constraint: /true|false/,
```

The end-of-input match is redundant, the code already wraps the regex in `^` and `$`